python-package/xgboost/spark/core.py

```diff
@@ -755,7 +758,12 @@ def _fit(self, dataset):
             k: v for k, v in train_call_kwargs_params.items() if v is not None
         }
         dmatrix_kwargs = {k: v for k, v in dmatrix_kwargs.items() if v is not None}
-        use_qdm = booster_params.get("tree_method", None) in ("hist", "gpu_hist")
+
+        if self.isDefined(self.use_quantile_dmatrix):
+            use_qdm = self.getOrDefault(self.use_quantile_dmatrix)
+        else:
+            use_qdm = CUDF_INSTALLED and \
```

Comment:
do we need to check CUDF in driver side?

Reply:
we can check CUDF in executor side, then different worker might have different use_qdm config if they have different python env.